model/src/main/kotlin/Hash.kt

```diff
@@ -19,68 +19,159 @@
 
 package com.here.ort.model
 
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.SerializerProvider
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer
+import com.fasterxml.jackson.databind.ser.std.StdSerializer
+
 import com.here.ort.utils.hash
 import com.here.ort.utils.toHexString
 
 import java.io.File
+import java.lang.reflect.InvocationTargetException
 import java.util.Base64
 
+import kotlin.reflect.full.primaryConstructor
+
 /**
- * A class that bundles a hash algorithm with its hash value.
+ * A sealed class of supported hash algorithms, providing the [names] a hash can be referred by (where the first name in
+ * the list is the default name), a [Regex][valueRegex] which all values have to match, the concrete hash [value], and a
+ * flag indicating whether values for this hash [can be calculated][canCalculate].
```

Comment:
I do like the model change because it bundles the algorithm and the hash in an object. But I do not see the benefit of merging the classes, having one class `Hash` to hold the data and one enum class `HashAlgorithm` to manage the algorithms seemed a better separation of concerns for me:
* It does not require reflection to iterate over the available algorithms.
* It does not require a custom serializer and implementations of `equals` and `hashCode` to "simulate" a data class.
* Similar simplifications in the calling code could have been achieved by adding a few helper functions, without merging the two classes.
* All other model classes are either data or enum classes, I would prefer if it stayed like this.

Reply:
Ok, I've reverted the merge into a sealed class and instead added the helper functions / refactored code like you propose.

Reply:
Thanks.